### PR TITLE
feat(ui): add French localization support

### PR DESCRIPTION
**Pull request by KikiManjaro**
## Description

Added localizations for the French language.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+    this.passwordIsRequiredErrorText = 'Un mot de passe est requis',
+    this.confirmPasswordIsRequiredErrorText = 'Veuillez confirmer votre mot de passe',
+    this.confirmPasswordDoesNotMatchErrorText = 'Les mots de passe de correspondent pas',
+    this.confirmPasswordInputLabel = 'Confirmez votre mot de passe',
+    this.forgotPasswordButtonLabel = 'Mot de passe oublié ?',
+    this.forgotPasswordViewTitle = 'Mot de passe oublié',
+    this.resetPasswordButtonLabel = 'Réinitialiser votre mot de passe',
+    this.verifyItsYouText = "Vérifiez votre identité",
+    this.differentMethodsSignInTitleText =
+        'Utiliseez une des méthodes suivantes pour vous connecter',
```
Comment (malbolged) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:223]: Utilisez une des méthodes suivantes pour vous connecter

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+        "Nous vous avons envoyé un email avec un lien pour réinitialiser votre mot de passe. Veuillez verifier vos emails.",
+    this.forgotPasswordHintText =
+        'Fournissez votre adresse email et nous vous enverrons un lien pour réinitialiser votre mot de passe',
+    this.emailLinkSignInButtonLabel = 'Connectez-vous avec le lien magique',
+    this.signInWithEmailLinkViewTitleText = 'Connectez-vous avec le lien magique',
+    this.signInWithEmailLinkSentText =
+        "Nous vous avons envoyé un email avec un lien magique. Verifiez vos emails et clickez sur le lien pour vous connecter",
+    this.sendLinkButtonLabel = 'Envoyer un lien magique',
+    this.arrayLabel = 'array',
+    this.booleanLabel = 'boolean',
```
Comment (malbolged) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:244]: booléen

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+    this.mapLabel = 'map',
+    this.nullLabel = 'null',
+    this.numberLabel = 'number',
```
Comment (malbolged) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:247]: numéro

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+    this.stringLabel = 'string',
```
Comment (malbolged) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:248]: chaîne de caractères

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+    this.typeLabel = 'type',
+    this.valueLabel = 'value',
```
Comment (malbolged) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:250]: valeur

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+    this.cancelLabel = 'cancel',
```
Comment (malbolged) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:251]: annuler

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+    this.updateLabel = 'update',
```
Comment (malbolged) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:252]: mettre à jour

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+    this.northInitialLabel = 'N',
+    this.southInitialLabel = 'S',
+    this.westInitialLabel = 'W',
```
Comment (malbolged) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:255]: O

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 9]
```diff
@@ -0,0 +1,263 @@
...
+    this.eastInitialLabel = 'E',
+    this.timestampLabel = 'timestamp',
+    this.longitudeLabel = 'longitude',
+    this.latitudeLabel = 'latitude',
+    this.geopointLabel = 'geopoint',
+    this.referenceLabel = 'reference',
```
Comment (malbolged) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:261]: référence

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 10]
```diff
@@ -0,0 +1,263 @@
...
+    this.stringLabel = 'string',
+    this.typeLabel = 'type',
+    this.valueLabel = 'value',
+    this.cancelLabel = 'cancel',
+    this.updateLabel = 'update',
+    this.northInitialLabel = 'N',
+    this.southInitialLabel = 'S',
+    this.westInitialLabel = 'W',
+    this.eastInitialLabel = 'E',
+    this.timestampLabel = 'timestamp',
```
Comment (malbolged) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:257]: horodatage

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 11]
```diff
@@ -0,0 +1,263 @@
...
+    this.longitudeLabel = 'longitude',
+    this.latitudeLabel = 'latitude',
+    this.geopointLabel = 'geopoint',
```
Comment (malbolged) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:260]: géopoint

**Review by malbolged — CHANGES_REQUESTED** [review]
@KikiManjaro Please fix these.
Thanks :)

**Comment by KikiManjaro** [comment]
Done ! :) Was not sure if the last ones should be translate, my bad

**Review by malbolged — APPROVED** [review]

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 12]
```diff
@@ -0,0 +1,263 @@
...
+    this.registerButtonText = 'Inscription',
+    this.linkEmailButtonText = 'Suivant',
+    this.signInWithPhoneButtonText = 'Se connecter avec un téléphone',
+    this.signInWithGoogleButtonText = 'Se connecter avec Google',
+    this.signInWithAppleButtonText = 'Se connecter avec Apple',
+    this.signInWithTwitterButtonText = 'Se connecter avec Twitter',
+    this.signInWithFacebookButtonText = 'Se connecter avec Facebook',
+    this.phoneVerificationViewTitleText = 'Entrez votre numéro de téléphone',
+    this.verifyPhoneNumberButtonText = 'Suivant',
+    this.verifyCodeButtonText = 'Verifier',
```
Comment (rrousselGit) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:186]:
```suggestion
    this.verifyCodeButtonText = 'Vérifier',
```

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 13]
```diff
@@ -0,0 +1,263 @@
...
+    this.verifyingPhoneNumberViewTitle = 'Entrez le code reçu par sms',
+    this.unknownError = 'Une erreur inconnue s\'est produite',
+    this.smsAutoresolutionFailedError =
+        'Impossible de retrouver le code automatiquement. Veuillez entrer votre code manuellement',
+    this.smsCodeSentText = 'Code envoyé par SMS',
+    this.sendingSMSCodeText = 'Envoi du code par SMS',
+    this.verifyingSMSCodeText = 'Verification du code SMS',
```
Comment (rrousselGit) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:193]:
```suggestion
    this.verifyingSMSCodeText = 'Vérification du code SMS',
```

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 14]
```diff
@@ -0,0 +1,263 @@
...
+    this.countryCode = 'Code',
+    this.codeRequiredErrorText = 'Un code de pays est requis',
+    this.invalidCountryCode = 'Code pays invalide',
+    this.chooseACountry = 'Choisissez un pays',
+    this.enableMoreSignInMethods = 'Activer plus de moyens de connexion',
+    this.signInMethods = 'Moyens de connexion',
+    this.provideEmail = 'Fournissez votre adresse email et votre mot de passe',
+    this.goBackButtonLabel = 'Retour',
+    this.passwordResetEmailSentText =
+        "Nous vous avons envoyé un email avec un lien pour réinitialiser votre mot de passe. Veuillez verifier vos emails.",
```
Comment (rrousselGit) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:235]:
```suggestion
        "Nous vous avons envoyé un email avec un lien pour réinitialiser votre mot de passe. Veuillez vérifier vos emails.",
```

File: packages/flutterfire_ui/lib/src/i10n/lang/fr.dart [hunk 15]
```diff
@@ -0,0 +1,263 @@
...
+    this.forgotPasswordHintText =
+        'Fournissez votre adresse email et nous vous enverrons un lien pour réinitialiser votre mot de passe',
+    this.emailLinkSignInButtonLabel = 'Connectez-vous avec le lien magique',
+    this.signInWithEmailLinkViewTitleText = 'Connectez-vous avec le lien magique',
+    this.signInWithEmailLinkSentText =
+        "Nous vous avons envoyé un email avec un lien magique. Verifiez vos emails et clickez sur le lien pour vous connecter",
```
Comment (rrousselGit) [inline packages/flutterfire_ui/lib/src/i10n/lang/fr.dart:241]:
```suggestion
        "Nous vous avons envoyé un email avec un lien magique. Vérifiez vos emails et clickez sur le lien pour vous connecter",
```

**Review by rrousselGit — APPROVED** [review]
LGTM

**Review by Salakar — APPROVED** [review]

**Pull request merged**